client/src/components/Masthead/QuotaMeter.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+    <current-user>
+        <div class="quota-meter d-flex align-items-center">
+            <b-link v-if="!hasQuota" v-b-tooltip.hover.left to="/storage" class="ml-auto quota-text" :title="title">
+                {{ usingString + " " + totalUsageString }}
+            </b-link>
+            <b-link v-else v-b-tooltip.hover.left class="quota-progress" to="/storage" :title="title">
+                <b-progress :value="usage" :max="100" :variant="variant" />
+                <span>{{ usingString + " " + usage.toFixed(0) }}%</span>
+            </b-link>
+        </div>
+    </current-user>
+</template>
+
+<script>
+import { bytesToString } from "utils/utils";
+import CurrentUser from "components/providers/CurrentUser";
+import { mapGetters } from "vuex";
+
+export default {
+    name: "QuotaMeter",
+    components: {
+        CurrentUser,
+    },
+    data() {
+        return {
+            usingString: this.l("Using"),
+        };
+    },
+    computed: {
+        ...mapGetters("config", ["config"]),
+        ...mapGetters("user", ["currentUser"]),
+        hasQuota() {
+            const quotasEnabled = this.config?.enable_quotas ?? false;
+            const quotaLimited = this.currentUser?.quota !== "unlimited" ?? false;
+
+            return quotasEnabled && quotaLimited;
+        },
+        usage() {
+            if (this.hasQuota) {
+                return this.currentUser?.quota_percent ?? 0;
+            } else {
+                return this.currentUser?.total_disk_usage ?? 0;
+            }
+        },
+        totalUsageString() {
+            const total = this.currentUser?.total_disk_usage ?? 0;
+            return bytesToString(total, true);
+        },
+        title() {
+            let details = "";
+            if (this.currentUser.isAnonymous) {
+                details = this.l("Log in for details.");
+            } else {
+                details = this.l("Click for details.");
+            }
+
+            return this.usingString + " " + this.totalUsageString + ". " + details;
```

Comment:
If no quota, we already display the exact usage directly so this can be redundant -- maybe only show "Click for details" in that case?

Reply:
Implemented in [bb6e245](https://github.com/galaxyproject/galaxy/pull/14435/commits/bb6e24596abe309595af621b857f3b1bc550e03c)